### PR TITLE
fix: ICal Attribute error & add integration test

### LINKED ISSUE #6511: ICalExporter not working
https://github.com/fossasia/open-event-server/blob/97e94f2ce14d20341253822b5456c93bef590c71/app/api/helpers/ICalExporter.py#L48-L53

There is no email attribute on event object.

Fix the method and write tests

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Fixes #6511 

#### Changes proposed in this pull request:

-
-
-

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/helpers/ICalExporter.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def export(event_id):
             for speaker in session.speakers:
                 # Ref: http://icalendar.readthedocs.io/en/latest/usage.html#file-structure
                 # can use speaker.email below but privacy reasons
-                attendee = vCalAddress('MAILTO:' + event.email if event.email else 'undefined@email.com')
```
Comment (iamareebjamal) [inline app/api/helpers/ICalExporter.py]: That's not what we want to do. People will get emails of speakers by exporting large events and use it to create spam list

That's the privacy issue with it.

Ans also, there needs to a test

Comment (mrsaicharan1) [reply app/api/helpers/ICalExporter.py]: Just to be clear, organizers are the only individuals who'd be able to export the events in ical formats right? Organizers would need the list of emails, wouldn't they?
Also, what email should be considered here? If we were to add an email attribute to the event, what should it be?

Comment (iamareebjamal) [reply app/api/helpers/ICalExporter.py]: If only organizers are able to export events, then it's not an issue.

If not, then owner email should be set

Comment (mrsaicharan1) [reply app/api/helpers/ICalExporter.py]: Currently users who organize events are the only individuals who can export them in `.ical` format. Therefore, there doesn't seem to be an issue here. Will test it and remove `DO-NOT-MERGE`

**Comment by mrsaicharan1** [comment]
@iamareebjamal I've added tests to check the presence of subcomponents in an `str` format.

**Comment by iamareebjamal** [comment]
It doesn't test for speaker, which is the failing case/

**Comment by mrsaicharan1** [comment]
@iamareebjamal I couldn't link the factories to test it. It always shows up as None. 
According to factoryboy docs,
Whenever I link the speaker with the session factory in the usual fashion, the `str` version of the exported ical shows absolutely no info pertaining to the speakers.
```
test_session = SessionFactory()
test_speaker = SpeakerFactory()
test_speaker.session = test_session
save_to_db(test_session)
save_to_db(test_speaker)
test_cal_str = icalexport_object.export(test_session.event_id)
```


Here is the string repr of the ical export
```
['BEGIN:VCALENDAR', 'VERSION:2.0', 'PRODID:-//fossasia//open-event//EN', 'X-WR-CALDESC:Schedule for sessions at example', 'X-WR-CALNAME:example', 'BEGIN:VEVENT', 'SUMMARY:example', 'DTSTART;VALUE=DATE-TIME:20991213T000000', 'DTEND;VALUE=DATE-TIME:20991214T000000', 'UID:1-4a766c6a', 'DESCRIPTION:example', 'GEO:1.23456789;1.23456789', 'LOCATION:example', 'URL:http://localhost/v1/events?identifier=4a766c6a', 'END:VEVENT', 'END:VCALENDAR', '']
```

**Comment by iamareebjamal** [comment]
Will test

**Pull request merged**